client/components/button-icon.vue

```diff
@@ -0,0 +1,63 @@
+<template>
+  <component
+    class="button-icon"
+    :href="href"
+    :is="tag"
+    :to="to"
+    @click="onClick"
+  >
+    <span class="icon" :class="icon" :style="{ 'font-size': size }" />
```

Comment:
Curious, did you try to update your existing button component to have optional icon? 
(Base Web does this with [startEnhancer/endEnhancer](https://baseweb.design/components/button/), maybe you could follow a similar pattern?)

Reply:
Yes your right. What I could do is put in a slot into the other component and use style overrides. I just thought that might be abit messier. I'll look into startEnhancer and perhaps raise a separate PR. Thanks